nixos/modules/services/monitoring/gatus.nix

```diff
@@ -0,0 +1,147 @@
+{ pkgs, lib, config, ... }:
+let
+  cfg = config.services.gatus;
+
+  settingsFormat = pkgs.formats.yaml { };
+
+  inherit (lib)
+    getExe
+    literalExpression
+    mdDoc
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOptionMD
+    types
+    ;
+in
+{
+  options = {
+    services.gatus = {
+      enable = mkEnableOption (mdDoc "Gatus");
+
+      package = mkPackageOptionMD pkgs "gatus" { };
+
+      user = mkOption {
+        type = types.str;
+        default = "gatus";
+        description = mdDoc "User account under which Gatus runs.";
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "gatus";
+        description = mdDoc "Group under which Gatus runs.";
+      };
+
+      configFile = mkOption {
+        type = with types; nullOr path;
+        default = null;
+        description = mdDoc ''
+          Path to the Gatus configuration file.
+          Overrides any configuration made using the `settings` option.
+        '';
+      };
+
+      environmentFile = mkOption {
+        type = with types; nullOr path;
+        default = null;
+        description = mdDoc ''
+          File to load as environment file.
+          Environmental variables from this file can be interpolated in the configuration file using `''${VARIABLE}`.
+          This is useful to avoid putting secrets into the nix store.
+        '';
+      };
+
+      settings = mkOption {
+        type = types.submodule {
+          freeformType = settingsFormat.type;
+          options = {
+            web.port = mkOption {
+              type = types.int;
+              default = 8080;
+              description = ''
+                The TCP port to serve the Gatus service at.
+              '';
+            };
+          };
+        };
+
+        default = { };
+
+        example = literalExpression ''
+          {
+            web.port = 8080;
+            endpoints = [{
+              name = "website";
+              url = "https://twin.sh/health";
+              interval = "5m";
+              conditions = [
+                "[STATUS] == 200"
+                "[BODY].status == UP"
+                "[RESPONSE_TIME] < 300"
+              ];
+            }];
+          }
+        '';
+
+        description = mdDoc ''
+          Configuration for Gatus.
+          Supported options can be found at the [docs](https://gatus.io/docs).
+        '';
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = mdDoc ''
+          Whether to open the firewall for the Gatus web interface.
+        '';
+      };
+    };
+  };
+
+  config =
+    let
+      settings = settingsFormat.generate "gatus.yaml" cfg.settings;
```

Comment:
you might want to move this as the default value for configFile, so you can remove a lot of checks and the weird indent
(if we decide to keep configFile, i'm not sure about that)